tools/fix_mailmap.py

```diff
@@ -0,0 +1,573 @@
+#!/usr/bin/env python3
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Check and fixup mailmaps files against git log."""
+
+from __future__ import annotations
+
+import argparse
+import re
+import subprocess
+import sys
+from enum import IntEnum, unique
+from pathlib import Path
+from sys import stderr
+from unicodedata import normalize
+
+# Set up some globals for simplicity
+# pylint: disable=invalid-name, global-statement
+color_output = True
+quiet_mode = False  # Only report errors
+problem_found = False  # Problems requiring user intervention were detected
+changes_made = False  # .mailmap was updated
+mailmap_path = Path()  # We'll set it correctly later
+mailmap_body_offset = 0  # Offset from the .mailmap header into the body
+
+
+@unique
+class ExitCode(IntEnum):
+    """Program exit codes enumeration."""
+
+    NO_PROBLEM = 0
+    CHANGES_MADE = 1
+    MAILMAP_NOT_FOUND = 2
+    MAILMAP_FORMAT = 3
+    GIT_LOG = 4
+    MAILMAP_NEEDS_MANUAL_CHANGE = 5
+
+
+def mailmap_line(index: int) -> str:
+    """Format a string to refer to a mailmap line in console messages."""
+    return f"{mailmap_path}:{mailmap_body_offset+index}"
+
+
+def print_red(text: str) -> None:
+    """Print messages indicating need for user intervention."""
+    global problem_found
+    problem_found = True
```

Comment:
Oh boy, that's quite the side effect lol.

Reply:
:-D